From 204a102966c2a8398f9845c2f36da7c59e1364ff Mon Sep 17 00:00:00 2001
From: Rahul Sandhu <nvraxn@gmail.com>
Date: Thu, 26 Jun 2025 11:30:05 +0100
Subject: [PATCH 2/2] autotools: use printf instead of echo -e for posix
 compliance

posix echo does not support the -e switch[1] causing systems with
/bin/sh pointing to a posix compliant shell to emit "-e" literally.

[1] https://pubs.opengroup.org/onlinepubs/9799919799/utilities/echo.html

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>
---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 3625717..45c7278 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -112,9 +112,9 @@ EXTRA_DIST += \
 	$(top_builddir)/icons/usbguard-icon.svg
 
 $(top_builddir)/src/usbguard-icon.hpp: $(top_builddir)/icons/usbguard-icon.svg
-	echo -e "#ifndef ICON_HPP\n#define ICON_HPP\nnamespace notify {\nconst char *icon =" > $@
+	printf "#ifndef ICON_HPP\n#define ICON_HPP\nnamespace notify {\nconst char *icon =" > $@
 	$(SED) 's/"/\\"/g' $^ | $(SED) 's/^/"/' | $(SED) 's/$$/\\n"/' >> $@
-	echo -e ";\n}\n#endif" >> $@
+	printf ";\n}\n#endif" >> $@
 
 CLEANFILES += \
 	$(top_builddir)/src/usbguard-icon.hpp
-- 
2.50.0

